code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -134,72 +134,81 @@
 
 /mob/living/carbon/human/adjustCloneLoss(amount)
 	if(dna.species && amount > 0)
-		amount = amount * dna.species.clone_mod
+		amount *= dna.species.clone_mod
 	. = ..()
 
-	var/heal_prob = max(0, 80 - getCloneLoss())
 	var/mut_prob = min(80, getCloneLoss() + 10)
-	if(amount > 0) //cloneloss is being added
-		if(prob(mut_prob))
-			var/list/obj/item/organ/external/candidates = list() //TYPECASTED LISTS ARE NOT A FUCKING THING WHAT THE FUCK
-			for(var/obj/item/organ/external/O in bodyparts)
-				if(O.is_robotic())
-					continue
-				if(!(O.status & ORGAN_MUTATED))
-					candidates |= O
-
-			if(candidates.len)
-				var/obj/item/organ/external/O = pick(candidates)
-				O.mutate()
-				to_chat(src, "<span class='notice'>Something is not right with your [O.name]...</span>")
-				O.add_autopsy_data("Mutation", amount)
-				return
+	var/heal_prob = max(0, 80 - getCloneLoss())
 
-	else //cloneloss is being subtracted
-		if(prob(heal_prob))
-			for(var/obj/item/organ/external/O in bodyparts)
-				if(O.status & ORGAN_MUTATED)
-					O.unmutate()
-					to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
-					return
+	if(getCloneLoss() == 0)  // All cloneloss was purged - fix all organs
+		for(var/obj/item/organ/external/O in bodyparts)
+			if(O.status & ORGAN_MUTATED)
+				O.unmutate()
+				to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
+	else if(amount > 0) // Cloneloss was inflicted - chance to mutate an organ
+		if(!prob(mut_prob))
+			return
 
+		var/list/candidates = list()
+		for(var/obj/item/organ/external/O in bodyparts)
+			if(!O.is_robotic() && !(O.status & ORGAN_MUTATED))
+				candidates |= O
+
+		if(length(candidates))
+			var/obj/item/organ/external/O = pick(candidates)
+			O.mutate()
+			to_chat(src, "<span class='notice'>Something is not right with your [O.name]...</span>")
+			O.add_autopsy_data("Mutation", amount)
+	else if (amount < 0) // Cloneloss was healed - chance to unmutate an organ
+		if(!prob(heal_prob))
+			return
 
-	if(getCloneLoss() < 1) //no cloneloss, fixes organs
 		for(var/obj/item/organ/external/O in bodyparts)
 			if(O.status & ORGAN_MUTATED)
 				O.unmutate()
 				to_chat(src, "<span class='notice'>Your [O.name] is shaped normally again.</span>")
+				return
```

Comment:
I cant suggest changes here because there are unmodified lines, but if you are applying the above it is ESSENTIAL you also pull this section up a level.

We can remove this `if` because if we check for 0 with guard clause, and check for more than 0 above, the only logical remaining option if it got this far is less than 0.

Also, please reply to let me know if it's intended that this only fixes one organ, I assume it is but want to be sure.

Reply:
As is, it's intended to be one per heal with a chance proportional to the clone damage post-adjustment, yeah; only when you purge all cellular damage does it heal all organs at once